components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAML2SSOFederatedAuthenticatorConfigBuilder.java

```diff
@@ -62,260 +80,261 @@ private static FederatedAuthenticatorConfig parse(EntityDescriptor entityDescrip
-                    if (entityDescriptor.getEntityID() != null && entityDescriptor.getEntityID().length()>0) {
-                        property.setValue(entityDescriptor.getEntityID());
-                    } else {
-                        property.setValue("");
-                        throw new IdentityApplicationManagementException("No Entity ID found, invalid file content");
-                    }
-                    properties[0] = property;
-
-                    property = new Property();
-                    property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID);
-                    property.setValue("");//not available in the metadata specification
-                    properties[1] = property;
-
-                    property = new Property();
-                    property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.SSO_URL);
-                    List<SingleSignOnService> singleSignOnServices = idpssoDescriptor.getSingleSignOnServices();
-                    if (CollectionUtils.isNotEmpty(singleSignOnServices)) {
-                        boolean found = false;
-                        for (int j = 0; j < singleSignOnServices.size(); j++) {
-                            SingleSignOnService singleSignOnService = singleSignOnServices.get(j);
-                            if (singleSignOnService != null) {
-                                if (singleSignOnService.getLocation() != null) {
-                                    property.setValue(singleSignOnService.getLocation());
-                                    found = true;
-                                    break;
-                                }
+                Property[] properties = new Property[24];
+
+                Property property = new Property();
+                property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.IDP_ENTITY_ID);
+                if (entityDescriptor.getEntityID() != null && entityDescriptor.getEntityID().length() > 0) {
+                    property.setValue(entityDescriptor.getEntityID());
+                } else {
+                    property.setValue("");
+                    throw new IdentityApplicationManagementException("No Entity ID found, invalid file content");
+                }
+                properties[0] = property;
+
+                property = new Property();
+                property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID);
+                property.setValue(""); // Not available in the metadata specification.
+                properties[1] = property;
+
+                property = new Property();
+                property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.SSO_URL);
+                List<SingleSignOnService> singleSignOnServices = idpssoDescriptor.getSingleSignOnServices();
+                if (CollectionUtils.isNotEmpty(singleSignOnServices)) {
```

Comment:
You don't need this `if`.

Reply:
This was a checkstyle fix. Will open separate PR for the fix and checkstyle issues.